src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -429,12 +432,57 @@ def autoregressive_inference(
         if shift_positions_by_one:
             positions -= 1
         input_ids = numpy.array([[new_token]])
-        engine_inputs = [input_ids, attention_mask, positions]
+        causal_mask = create_causal_mask(input_ids, attention_mask)
+        engine_inputs = [input_ids, attention_mask, positions, causal_mask]
 
         generated_token, generated_logits = self.engine(engine_inputs)
 
         return generated_token, generated_logits
 
+    def engine_inputs_for_prefill(
+        self, tokens: List[int]
+    ) -> Generator[List[numpy.ndarray], None, None]:
+        """
+        Takes a list of tokens and turns the first
+        `self.prompt_processing_sequence_length` tokens into
+        appropriate engine inputs for the multitoken engine.
+
+        :param tokens: the list of tokens to process
+        :return: a generator of engine inputs
+        """
+        # TODO: Make it yield multiple engine_inputs
+        # Once this is done, we can update the docstrings and be
+        # more verbose about what this function does
```

Comment:
can you add a few more in-line comments to this function especially the logic branches?
potentially expand the docstring as well?

Also is this function to support multitoken or causal mask? checking understanding as this is the causal mask PR (good either way)